File: csgo-leagues-ios/csgo-leagues-ios/Domain/Entities/Match.swift
import Foundation

struct Match: Equatable, Hashable, Identifiable, Sendable {
    let id: Int
    let name: String
    let status: MatchStatus
    let beginAt: Date?
    let teams: [Team]       // 0, 1, or 2 teams
    let league: League
    let serieName: String

    var team1: Team? { teams[safe: 0] }
    var team2: Team? { teams[safe: 1] }
}

File: csgo-leagues-android/app/src/main/java/com/voidx/csgo/domain/entity/Match.kt
package com.voidx.csgo.domain.entity

import java.time.LocalDateTime
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable

@Serializable
data class Match(
    val id: Int,
    val name: String,
    val status: MatchStatus,
    @Contextual val beginAt: LocalDateTime? = null,
    val teams: List<Team> = emptyList(),
    val league: League,
    val serieName: String,
)
